packages/block-editor/src/components/inserter/library.js

```diff
@@ -36,6 +38,14 @@ function InserterLibrary( {
 		[ clientId, rootClientId ]
 	);
 
+	const inserterRef = useRef();
+	useEffect( () => {
+		if ( ! shouldFocusSearch ) {
+			return;
+		}
+		inserterRef.current.searchFocus();
+	}, [ shouldFocusSearch ] );
```

Comment:
This effect is misplaced, it should remain where the existing effect is because the library is not necessarily a popover which means it doesn't need to move the focus in all usage.